app/routers/event.py

```diff
@@ -323,6 +332,63 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
         return None
 
 
+def extract_shared_list_from_data(data: ImmutableMultiDict,
+                                  db: Session) -> Optional[SharedList]:
+    """Extract shared list items from POST data.
+    Return:
+        SharedList: SharedList object stored in the database.
+    """
+
+    Item = namedtuple("Item", ["name", "amount", "participant"])
+    shared_list_raw = {
+        'item_name': data.getlist('item_name'),
+        'item_amount': data.getlist('item_amount'),
+        'item_participant': data.getlist('item_participant')
+    }
+    shared_list = {"title": data.get("title"),
+                   "items": []}
+    for i in range(len(shared_list_raw.get('item_name'))):
+        try:
+            item = Item(
+                name=shared_list_raw["item_name"][i],
+                amount=shared_list_raw["item_amount"][i],
+                participant=shared_list_raw["item_participant"][i]
+            )
+            if _check_item_is_valid(item):
+                item_dict = item._asdict()
+                item_dict["amount"] = float(item_dict["amount"])
+                shared_list["items"].append(item_dict)
+        except (KeyError, IndexError):
+            continue
```

Comment:
Please don't do that.

1. Catch exception only for the relevant line.
2. Don't use continue to silence errors.
3. Catch only the relevant errors that may arise for the relevant section of code.